doc/operation-and-maintenance/Mongoose-metrics.md

```diff
@@ -1,149 +1,217 @@
-| xmppMessageReceived | spiral | host, XMPP | Number of messages received by the server |
-| xmppMessageSent | spiral | host, XMPP | Number of messages sent to clients |
-| xmppPresenceReceived | spiral | host, XMPP | Number of presences received by the server |
-| xmppPresenceSent | spiral | host, XMPP | Number of presences sent to clients |
-| xmppStanzaCount | spiral | host, XMPP | Number of all stanzas sent to clients |
-| xmppStanzaDropped | spiral | host, XMPP | Number of dropped stanzas |
-| xmppStanzaReceived | spiral | host, XMPP | Number of stanzas received by the server|
-| xmppStanzaSent | spiral | host, XMPP | Numb of stanzas sent to clients|
-
-Metrics assgined to a group `hook` are generic metrics updated when a given hook is run. 
-The names of these metrics are the same as a corresponding hook name.
-
-Most of `XMPP` metrics are also triggered by hook and their count is the same as the corresponding hook's runs. 
-They are named differently to maintain backward compatibility.
-
-To see the `hook <-> XMPP metric` translation please refer to [mongoose_metrics_hooks](https://github.com/esl/MongooseIM/blob/exometer/apps/ejabberd/src/mongoose_metrics_hooks.erl#L71) file.
- 
-## Global metrics description
+    * Hook metrics.
+    They are created for every [hook](../developers-guide/Hooks-and-handlers.md) and incremented on every call to it.
+* Global metrics: Metrics common for all XMPP hosts.
+    * Data metrics.
+    These are misc. metrics related to data transfers (e.g. sent and received stanza size statistics).
+    * VM metrics. Basic Erlang VM statistics.
+* Backend metrics: Histograms with timings of calls to various backends.
 
-| Metric name | Type | Description |
-| ----------- | ---- | ----------- |
-| nodeSessionCount | function | Number of sessions on a given Erlang node |
-| totalSessionCount | function | Total number of sessions in the cluster |
-| uniqueSessionCount | function | Number of unique sessions in the cluster |
+## Metrics types
+
+### `spiral`
+
+This kind of metric provides 2 values: `total` event count (e.g. stanzas processed) and a value in 60s window (`one` value).
+Dividing `one` value by 60 provides an average per-second value over last minute.
+
+**Example:** `[{total, 1000}, {one, 20}]`
+
+### `value`
+
+A simple value.
+It is actually a one-element proplist: `[{value, N}]`.
+
+**Example:** `[{value, 256}]`
+
+### `gauge`
+
+It is similiar to a `value` type but consists of two properties:
+
+* `value`
+* `ms_since_reset` - Time in milliseconds elapsed from last metric update.
+
+**Example:** `[{value, 12}, {ms_since_reset, 91761}]`
+
+### `proplist`
+
+A metric which is a nonstandard proplist.
+You can find the lists of keys in metrics descriptions.
+
+**Example:** `[{total,295941736}, {processes_used,263766824}, {atom_used,640435}, {binary,1513152}, {ets,3942592}, {system,32182072}]`
+
+### `histogram`
+
+A histogram collects values over sliding window of 60s and exposes following stats:
```

Comment:
A histogram collects values over a sliding window of 60s and exposes the following stats: